src/preload.js

```diff
@@ -22,6 +28,26 @@ function init() {
   window.MakerBridge = {
     getVersion,
   };
+  window.flashBoardFirmware = flashBoardFirmware;
+}
+
+function flashBoardFirmware(boardType) {
+  if (boardType === BOARD_TYPE.CLASSIC) {
```

Comment:
I'm a little wary of trusting the caller of this function to provide the board type.  Is this something `flashBoardFirmware` should determine (or at least verify) itself before it tries to make changes to the board?

Reply:
I'm curious as to why you are wary of this, but yes, this is something we can calculate in the maker browser. However, to do it from scratch duplicates code that is already used in the cdo-repo. Would splitting the difference and passing in a port object to `flashBoardFirmware` and then comparing that against the expected product/vendor of the Classic be an appropriate measure?

Reply:
Ahh I see what you mean! To continue to poke at this, rather than passing a board type or port, we could pass an object with all the information needed to flash the board. Example, the `flashBoardFirmware` take an object parameter that looks like `{name: 'circuit-playground-class', hexPath: 's3.code.org.path-to-hex-file', checksum: checksum}`. Then all the logic for `if(CPC){flash}` sits in the cdo repo and the only thing `flashBoardFirmware` knows is 'you gave me a board to flash and file to flash it with'.

Reply:
Oooh, I like then.  The nice thing about that is if we want to add support for flashing different boards in the future, we don't have to ship a Maker App update - just include it in our daily release.